src/content/en/2025/generative-ai.md

```diff
@@ -1,20 +1,413 @@
+
+The <a hreflang="en" href="https://webmachinelearning.github.io/">World Wide Web Consortium (W3C) Web Machine Learning</a> Community Group (WebML CG) and Working Group (WebML WG) are actively standardizing this shift to make AI a first-class citizen of the web. This effort follows two primary architectural directions: *Bring Your Own AI* and *Built-in AI*.
+
+#### Bring Your Own AI
+
+In the Bring Your Own AI (BYOAI) model, the developer is responsible for shipping the model to the user. The web application downloads a specific model binary and executes it using low-level APIs on local hardware. This allows running highly specialized models, tailored to the use case, but also requires significant bandwidth.
+
+There are <a hreflang="en" href="https://www.w3.org/2024/01/webevolve-series-events/media/slides/hu-ningxin.pdf#page=4">three processing units</a> that can be used to run AI inference locally:
+
+* Central Processing Unit (CPU): Responds quickly, ideal for low-latency AI workloads.
+* Graphics Processing Unit (GPU): High throughput, ideal for AI-accelerated digital content creation.
+* Neural Processing Unit (NPU) or Tensor Processing Unit (TPU): Low power, ideal for sustained AI workloads and AI offload for battery life.
+
+The three key APIs that facilitate local AI inference are WebAssembly (CPU), WebGPU (GPU), and WebNN (CPU, GPU, and NPU).
+
+All statistics in this section refer to the percentage of Google Chrome page loads that utilize these APIs at least once, across all release channels and platforms. It is essential to note that the use of WebAssembly and WebGPU does not confirm AI activity; these are general-purpose APIs frequently utilized for tasks such as complex calculations, 3D visualizations, or gaming.
+
+##### WebAssembly
+
+[WebAssembly](https://developer.mozilla.org/en-US/docs/WebAssembly) (Wasm) acts as the bytecode for the web. Code written in various programming languages, including C++ and Rust, can be compiled into WebAssembly. It enables developers to write optimized, high-performance code that is executed by the browser's scripting engine on the user's CPU.
+
+<!-- markdownlint-disable-next-line MD051 -->
+WebAssembly has broad browser support, being implemented in all relevant browser engines since 2017. In 2025, the usage of WebAssembly (i.e., the instantiation of a WebAssembly module relative to page loads) experienced 27% linear growth (see [Figure 1](#fig-1)) from being active on 4.44% of page loads in January, and 5.64% in December 2025 (January 2024: 3.37%).
```

Comment:
Technically it's figure 7.1, but you might be better saying something like "see the following figure" as much less likely to get out of sync if you add or remove figures.


```suggestion
WebAssembly has broad browser support, being implemented in all relevant browser engines since 2017. In 2025, the usage of WebAssembly (i.e., the instantiation of a WebAssembly module relative to page loads) experienced 27% linear growth (see [Figure 7.1](#fig-1)) from being active on 4.44% of page loads in January, and 5.64% in December 2025 (January 2024: 3.37%).
```